sympy/vector/coordsysrect.py

```diff
@@ -98,6 +213,65 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            if parent is None:
```

Comment:
This `if` loop is not very useful. The case where `parent == None` and `parent != None` could be handled by the same code.